libs/features/mas/web-components/src/hydrate.js

```diff
@@ -0,0 +1,123 @@
+import { createTag } from './utils.js';
+
+export async function hydrate(fragmentData, merchCard) {
+    const fragment = fragmentData.fields.reduce(
+        (acc, { name, multiple, values }) => {
+            acc[name] = multiple ? values : values[0];
+            return acc;
+        },
+        { id: fragmentData.id },
+    );
+    const { variant } = fragment;
+    if (!variant) return;
+    fragment.model = fragment.model;
+
+    merchCard.variant = variant;
+    await merchCard.updateComplete;
+    const { aemFragmentMapping } = merchCard.variantLayout;
+
+    if (!aemFragmentMapping) return;
+
+    merchCard.variantLayout.refs ??= [];
+    merchCard.variantLayout.refs.forEach((ref) => ref.remove());
+    const appendFn = (el) => {
+        merchCard.variantLayout.refs.push(el);
+        merchCard.append(el);
+    };
+
+    fragment.mnemonicIcon?.forEach((icon, idx) => {
+        const href =
+            fragment.mnemonicLink?.length > idx
+                ? fragment.mnemonicLink[idx]
+                : '';
+        const alt =
+            fragment.mnemonicAlt?.length > idx ? fragment.mnemonicAlt[idx] : '';
+        const merchIcon = createTag('merch-icon', {
+            slot: 'icons',
+            src: icon,
+            alt,
+            href,
+            size: 'l',
```

Comment:
CCD cards use a smaller size for the icons (s, 30x30px). I think we should add the option to set the size in M@S studio. Same thing goes for the CTAs

Reply:
let's address this in a future PR since this is not new code but simply logic moved to another file.